webpack_loader/templatetags/webpack_loader.py

```diff
@@ -7,15 +7,22 @@
 register = template.Library()
 
 
-@register.simple_tag
-def render_bundle(bundle_name, extension=None, config='DEFAULT', suffix='', attrs='', is_preload=False):
+@register.simple_tag(takes_context=True)
+def render_bundle(context, bundle_name, extension=None, config='DEFAULT', suffix='', attrs='', is_preload=False, skip_common_chunks=False):
     tags = utils.get_as_tags(
         bundle_name, extension=extension, config=config,
         suffix=suffix, attrs=attrs, is_preload=is_preload
     )
+    used_tags = context.get("webpack_loader_used_tags", [])
+    if not used_tags:
+        context["webpack_loader_used_tags"] = []
+    if skip_common_chunks:
+        tags = [mark_safe(tag) for tag in tags if tag not in used_tags]
```

Comment:
Is this `mark_safe` necessary?

Reply:
I'll have time to check later but I added it because nothing was getting set in the context without it. I think they get get removed by the autoescaping. 

> If the render() method of your template tag stores the result in a context variable (rather than returning the result in a string), it should take care to call mark_safe() if appropriate. When the variable is ultimately rendered, it will be affected by the auto-escape setting in effect at the time, so content that should be safe from further escaping needs to be marked as such.

https://docs.djangoproject.com/en/3.2/howto/custom-template-tags/#auto-escaping-considerations

Reply:
Turns out that no, it isn't necessary.